src/Uno.UI/UI/Xaml/DependencyPropertyDetailsCollection.Bindings.cs

```diff
@@ -84,11 +101,15 @@ internal void SuspendBindings()
 			{
 				_bindingsSuspended = true;
 
-				var bindings = _bindings.Data;
-
-				for (int i = 0; i < bindings.Length; i++)
+				var bindings = _bindings;
+				if (bindings is not null)
 				{
-					bindings[i].SuspendBinding();
+					bindings.Freeze();
```

Comment:
I like the idea, though it's not the same as having an immutable list, particularly if for some reason the property notification is reentrant. In such a case, handlers would be missed. Have you tried throwing of a list is already frozen and tried to be frozen again?

Reply:
@jeromelaban Would having a counter resolve your concern?
i.e, on Freeze we will increase the counter by one, and on Unfreeze we will decrease it, and we only push the queue to the list if the counter became zero?

i.e,

```
Count is 0 initially
Freeze: Count 1
Freeze: Count 2
Unfreeze: Count 1 // do nothing.
Unfreeze: Count 0 // push the queue to the actual list.
```

Reply:
That's where I'm not sure. In theory, the reentrant call (if that can happen) would invoke the new list, regardless of upper-stack freezing. That's where I'm asking if you throw when there's a reentrant call, does it actually happen with all the tests we have.

Reply:
@jeromelaban I added a `throw`. There are some test failures but they are not showing the exception, so I think the exception was caught somewhere and just causes the test to produce wrong results. I think I'll need to debug locally and see exactly what's happening. The hard part is understanding and finding a test case that will produce different results per the used approach.